### PR TITLE
(feat) Adding unit to measurments + fixing picking on the measurment views + Renaming

### PR DESCRIPTION
**Pull request by nilscognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:
1. Rename Tooltip to TranslateKey 
2.  Add unique index in  BaseCommand  to forces rerender in react for special rare cases.
3. Added CopyToClipboardCommand and  DeleteDomainObjectCommand, and used them in the DomainObjectPanel
4. Added ToogleMetricUnitsCommand for switching between metric and none metric units.
5. Added more comments around in the base of the architecture
6. Added unitSystem to the RootDomainObject , so units are handled one place
7. Rename NumberType  -> Quantity
8.  Refactor of PanelInfo to hadle the new unit system
9. Added class UnitSystem  to handle very basic unit system. This should be extended in the future.
10. Moved class ClosestGeometryFinder from Fusion. This is not deleted on the fusion side, but should when Annotations are using the new architecture
11. Added unit change in Changes and used when it was necessary
12. Use unit system in the measurement views and in the DomainObjectPanel'
13. Implement intersectIfCloser in MeasureLineView because the tree.js implementation did not work properly. 
14. Update the example code with the new unit system
15. Some refactoring in the CommandButtons.cs
16. Remove canBeChecked in commands. Not used.

TODO: The axis no not yet understand the unit system. It should. Will take a couple of hours.

## How has this been tested? :mag:

See below

## Test instructions :information_source:

1. Use the measurement tool to create some measurements
2. Note the new command with the measurement icon
3. Toggle this on and off. See the difference

image

Icons and the placement is discussed with Designer.

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts [hunk 1]
```diff
@@ -82,7 +94,57 @@ export class MeasureLineView extends GroupThreeView {
...
+    // returns out if any of those objects are intersected. Same for Boxes. Now the intersection is somewhat arbitrarly
+    // if style.depthTest == false. Then the last one added to the viewer will be picked first,
+    // regardless of the distance to the mouse.
+    if (style.depthTest && closestDistance !== undefined) {
+      closestFinder.minDistance = closestDistance;
+    }
+    const loopLength = domainObject.measureType === MeasureType.Polygon ? length + 1 : length;
+    for (let i = 0; i < loopLength; i++) {
+      thisPoint.copy(points[i % length]);
+      thisPoint.applyMatrix4(CDF_TO_VIEWER_TRANSFORMATION);
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts:128]: Are the points stored in CDF-coordinates?

File: react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts [hunk 2]
```diff
@@ -82,7 +94,57 @@ export class MeasureLineView extends GroupThreeView {
...
+
+      if (i > 0) {
+        const distanceSq = ray.distanceSqToSegment(prevPoint, thisPoint, undefined, intersection);
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts:131]: Should this be an early return?

**Review by haakonflatval-cognite — APPROVED** [review]
A couple of comments, but otherwise LGTM :+1:

File: react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts (hunk 1 above)
Comment (nilscognite) [reply react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts:128]: Yes

File: react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts (hunk 2 above)
Comment (nilscognite) [reply react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts:131]: Fixed

**Pull request merged**